ui/components/app/nft-details/nft-details.js

```diff
@@ -94,10 +97,26 @@ export default function NftDetails({ nft }) {
     'M/d/y',
   );
 
-  const onRemove = () => {
-    dispatch(removeAndIgnoreNft(address, tokenId));
-    dispatch(setNewNftAddedMessage(''));
-    dispatch(setRemoveNftMessage('success'));
+  const onRemove = async () => {
+    try {
+      await dispatch(removeAndIgnoreNft(address, tokenId));
+      dispatch(setNewNftAddedMessage(''));
+      dispatch(setRemoveNftMessage('success'));
+    } catch (err) {
+      dispatch(setNewNftAddedMessage(''));
+      dispatch(setRemoveNftMessage('error'));
+    }
+    // track event
+    trackEvent({
+      event: MetaMetricsEventName.NFTRemoved,
+      category: 'Wallet',
+      sensitiveProperties: {
```

Comment:
not sure how the schema of these events work, but a chain id seems like it would be useful

Reply:
totally